packages/slice-machine/src/features/models/ModelProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { CustomType } from "@prismicio/types-internal/lib/customtypes";
+import { PropsWithChildren, createContext, useContext, useMemo } from "react";
+
+type ModelContextValue =
+  | { type?: never; model?: never }
+  | { type: "customType"; model: CustomType }
+  | { type: "slice"; model: ComponentUI };
+
+const ModelContext = createContext<ModelContextValue>({});
+
+type ModelProviderProps = PropsWithChildren<
+  | { type?: never; model?: never }
+  | { type: "customType"; model: CustomType }
+  | { type: "slice"; model: ComponentUI }
+>;
+
+export function ModelProvider(props: ModelProviderProps) {
```

Comment:
Since the data is strictly used for tracking, let's keep it simple and use the route to determine the custom type. The provider is too heavy-handed for this purpose and adds complexity.

Reply:
Commit: https://github.com/prismicio/slice-machine/pull/1335/commits/0b01b237e4f3935716315634376fd46ead83c85a